frontend/src/scenes/experiments/experimentLogic.tsx

```diff
@@ -943,6 +947,16 @@ export const experimentLogic = kea<experimentLogicType>([
     })),
     selectors({
         props: [() => [(_, props) => props], (props) => props],
+        dynamicFeatureFlagKey: [
+            (s) => [s.experiment],
+            (experiment: Experiment): string => {
+                return experiment.name
```

Comment:
One thing to consider is that experiment creation will fail if there's already a flag with the same key. This might happen if users reuse common names like "Homepage experiment."

A possible workaround would be to add a random string to the end of the generated flag key, but it's probably not necessary, as I don't think conflicts will happen often.

In this case, the user will see an error toast about the conflict and will need to modify the key, which I think is fine.

Reply:
> In this case, the user will see an error toast about the conflict and will need to modify the key, which I think is fine.

Yep, and this is consistent with the existing behavior which I think is fine for now.